[DOCS] DOC-285 new guide: how to use self initializing expectations

Rachel-Reverie opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- add documentation on how to use a self initializing Expectation
- add example script under test for how to determine if an Expectation is self initializing
- add example script under test for how to use a self initializing Expectation
-


After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.


Thank you for submitting!

Rachel-Reverie committed:
- put example script for determining if an Expectation works under the self-initializing framework under test.

Rachel-Reverie committed:
- put example script for using a self-initializing Expectation under test.

Rachel-Reverie committed:
- incomplete draft of documentation update.

Rachel-Reverie committed:
- added example scripts under test.

Rachel-Reverie committed:
- minor update to path of data used in scripts under test

Rachel-Reverie committed:
- Adds snippet tags to scripts
- Finishes the how to guide contents

Rachel-Reverie committed:
- Adds remaining technical term tags to how-to guide
- Adds how-to guide to ToC

Rachel-Reverie committed:
-corrections to script docstrings for accuracy

Rachel-Reverie committed:
- update terminology (self-initializing to auto-initializing)

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5205&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5205&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/5205/82c27188/08cf4117cba3ebf695ffe9ce54a18cc63677cc0f.svg)](https://app.codesee.io/r/reviews?pr=5205&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-285
- update terminology in content and file names for docs from "self-initializing" to "auto-initializing"

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

Rachel-Reverie committed:
DOC-281
- updates broken link

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations

github-actions[bot] committed:
Merge develop into DOCS/DOC-278/DOC-281/how_to_use_self_initializing_expectations
